src/en/charms-storage.md

```diff
@@ -164,60 +172,72 @@ For example, as described above, attaching an EBS volume to an EC2 instance
-dynamically attached to a unit using `juju add-storage`.
+### Adding and detaching storage
 
-For example, to create a 100GiB EBS volume and attach it to unit 'postgresql/0'
-as its pgdata storage:
-
-```bash
-juju add-storage postgresql/0 pgdata=ebs,100G
-```
+Assuming the storage provider supports it, storage can be created and attached
+to a unit using `juju add-storage`. Juju will ensure the storage is allowed to
+attach to the unit's machine.
 
-Juju will ensure the storage is allowed to attach to the unit's machine. In the
-above example, the EBS volume was created in the same availability zone as the
-instance (a requirement).
+!!! Note:
+    Currently, it is not possible to add storage to the model without also
+    attaching it to a unit.
 
 Charms can specify a maximum number of storage instances. In the case of the
-postgresql charm, a maximum of one is allowed for 'pgdata'. If an attempt is
+charm 'postgresql', a maximum of one is allowed for 'pgdata'. If an attempt is
 made to exceed it, Juju will return an error.
 
 Dynamic storage can be detached from units using `juju detach-storage`.
 
-For example, to detach storage 'pgdata/0' from unit 'postgresql/0':
-
-```bash
-juju detach-storage postgresql/0 pgdata/0
-```
-
 Charms can also define a minimum number of storage instances. The postgresql
 charm specifies a minimum of zero for 'pgdata' whereas another charm may specify
 a different number. In any case, if detaching storage from a unit would bring
 the total number of storage instances below the minimum, Juju will return an
 error.
 
-#### Persistence
+#### Examples
+
+To create a 32GiB EBS volume and attach it to unit 'ceph-osd/0' as its OSD
+storage:
+
+```bash
+juju add-storage ceph-osd/0 osd-devices=ebs,32G,1
+```
+
+Above, the volume was created in the same availability zone as the instance (a
+requirement).
 
-Detaching storage from a unit does not destroy the storage. When a unit is
-removed from the model, and the unit has dynamic storage attached, Juju will
-detach the storage but leave it intact at the model level. This enables the
-storage to be re-attached to another unit using `juju attach-storage`, or to a
-new unit using the `--attach-storage` flag of `juju deploy` or `juju add-unit`:
+To detach OSD device 'osd-devices/2' from a Ceph unit:
 
-Detached storage can be destroyed and removed from the model using `juju remove-storage`.
+```bash
+juju detach-storage osd-devices/2
+```
+
+!!! Important:
+    Detaching storage from a unit does not destroy the storage.
+
+### Persistence
+
+As we saw, detaching storage does not destroy the storage. In addition, when a
+unit is removed from a model, and the unit has dynamic storage attached, the
+storage will be detached and left intact. This allows detached storage to be
+re-attached to an existing unit using `juju attach-storage`, or to a new unit
+using the `--attach-storage` flag of `juju deploy` or `juju add-unit`:
```

Comment:
It's collateral damage caused by a refactoring of the text that didn't get swept up by a thoroughgoing sub-review. I will replace it by a full stop.